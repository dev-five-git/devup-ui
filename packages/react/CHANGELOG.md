# @devup-ui/react

## 1.0.24

### Patch Changes

- [#459](https://github.com/dev-five-git/devup-ui/pull/459) [`840553d`](https://github.com/dev-five-git/devup-ui/commit/840553d3e3ddcdc901af99daa9ca87eca3f5a640) Thanks [@owjs3901](https://github.com/owjs3901)! - Global css typing

## 1.0.23

### Patch Changes

- [#449](https://github.com/dev-five-git/devup-ui/pull/449) [`b85123f`](https://github.com/dev-five-git/devup-ui/commit/b85123fce0c097649bd602566cb3601107c190f6) Thanks [@owjs3901](https://github.com/owjs3901)! - Support selector with params

## 1.0.22

### Patch Changes

- [#435](https://github.com/dev-five-git/devup-ui/pull/435) [`8e46b5e`](https://github.com/dev-five-git/devup-ui/commit/8e46b5eb35897ff33b27738f4623ca7bee2588fb) Thanks [@owjs3901](https://github.com/owjs3901)! - Export TypographyKeys

## 1.0.21

### Patch Changes

- [#426](https://github.com/dev-five-git/devup-ui/pull/426) [`3c65364`](https://github.com/dev-five-git/devup-ui/commit/3c65364125cea6e3582562b99a9b71291fc6f8c2) Thanks [@owjs3901](https://github.com/owjs3901)! - Support query on imports

## 1.0.20

### Patch Changes

- [#400](https://github.com/dev-five-git/devup-ui/pull/400) [`fee97f8`](https://github.com/dev-five-git/devup-ui/commit/fee97f8a3d79b9a4c62858deb8e1aea8c609e3a2) Thanks [@owjs3901](https://github.com/owjs3901)! - Add positioning

## 1.0.19

### Patch Changes

- 84d0c1f: divide children prop in props prop

## 1.0.18

### Patch Changes

- f7ee823: Support props and custom component in as

## 1.0.17

### Patch Changes

- 554b56e: Update useTheme

## 1.0.16

### Patch Changes

- 6c25c5d: Add maskPos, maskImg

## 1.0.15

### Patch Changes

- b6b2fed: Support \_ selector and typing in globalCss

## 1.0.14

### Patch Changes

- c42d636: Impl font-faces to global

## 1.0.13

### Patch Changes

- 18bc846: Fix nested selector typing issue

## 1.0.12

### Patch Changes

- 7dba3e9: Optimize typing

## 1.0.11

### Patch Changes

- 39895fc: Implement Dynamic props by csstype

## 1.0.10

### Patch Changes

- d3fa982: Remove reset css

## 1.0.9

### Patch Changes

- 132e8dd: Add globalCss

## 1.0.8

### Patch Changes

- c27d60b: Support source map

## 1.0.7

### Patch Changes

- f6d3f0a: Support nested selector

## 1.0.6

### Patch Changes

- 6182ae8: Fix styleVars type

## 1.0.5

### Patch Changes

- 2dd81f5: Implement styleVars

## 1.0.4

### Patch Changes

- 9d45acb: Update wasm

## 1.0.3

### Patch Changes

- c5336c9: Add writingMode, textOrientation

## 1.0.2

### Patch Changes

- bb2f49f: Add vendor properties (moz, webkit, ms)

## 1.0.1

### Patch Changes

- 864c37b: Add info to package.json

## 1.0.0

### Major Changes

- 18b1bd7: Optimize Zero value

## 0.1.34

### Patch Changes

- c029a8d: Fix theme script

## 0.1.33

### Patch Changes

- 57db495: Add initTheme, Fix typing issue of ThemeScript

## 0.1.32

### Patch Changes

- 5f2ad1d: Add default theme

## 0.1.31

### Patch Changes

- e7fd185: Add textDecoration type

## 0.1.30

### Patch Changes

- 0e1ab6e: Support for conditional selector

## 0.1.29

### Patch Changes

- 7535643: Change default generic type of VStack

## 0.1.28

### Patch Changes

- 98c7b47: Add support conditional pattern

## 0.1.27

### Patch Changes

- 68f79d0: Add lineClamp, willChange
- 456cb42: Add visiblity

## 0.1.26

### Patch Changes

- 27c414d: Implement style order

## 0.1.25

### Patch Changes

- a3684c5: Add selectors

## 0.1.24

### Patch Changes

- bf47284: Add borderTopRadius, borderBottomRadius, borderLeftRadius, borderRightRadius, \_print

## 0.1.23

### Patch Changes

- 2102fd7: Support theme typing to css

## 0.1.22

### Patch Changes

- e1d8d8b: Fix typing issue

## 0.1.21

### Patch Changes

- 96cd069: Fix theme typing issue

## 0.1.20

### Patch Changes

- 064476f: Fix TypographyKey type definition for better type checking

## 0.1.19

### Patch Changes

- 7e54c1f: Fix type checking issue

## 0.1.18

### Patch Changes

- 77afb9e: Support generic to as prop

## 0.1.17

### Patch Changes

- ea622ed: Add svg props

## 0.1.16

### Patch Changes

- e072762: convert css

## 0.1.15

### Patch Changes

- b40d274: Add typograhpy to props

## 0.1.14

### Patch Changes

- 7f19bd5: Implement ThemeScript, useTheme, getTheme, setTheme

## 0.1.13

### Patch Changes

- 00b14ca: Support optional args in css

## 0.1.12

### Patch Changes

- b84cbc1: Update css.ts

## 0.1.11

### Patch Changes

- 76e8fa1: Add grid, Support for selector in css

## 0.1.10

### Patch Changes

- 0da9456: Add content to style props

## 0.1.9

### Patch Changes

- 6f7c55d: Add group selector

## 0.1.8

### Patch Changes

- 8943bad: Add double selector

## 0.1.7

### Patch Changes

- 0040445: Fix selector type

## 0.1.6

### Patch Changes

- b25c769: Add placehoder to selector

## 0.1.5

### Patch Changes

- e9d655f: Add shorthand

## 0.1.4

### Patch Changes

- 7bde1f2: Add boxSizing prop

## 0.1.3

### Patch Changes

- 30443ae: Update readme

## 0.1.2

### Patch Changes

- 242a6c2: Update package

## 0.1.1

### Patch Changes

- 6100644: Add Image component

## 0.1.0

### Minor Changes

- c0ff96f: Deploy
