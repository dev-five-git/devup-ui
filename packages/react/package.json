{
  "name": "@devup-ui/react",
  "description": "Zero Config, Zero FOUC, Zero Runtime, CSS in JS Preprocessor",
  "repository": "https://github.com/dev-five-git/devup-ui",
  "author": "devfive",
  "license": "Apache-2.0",
  "homepage": "https://devup-ui.com",
  "bugs": {
    "url": "https://github.com/dev-five-git/devup-ui/issues",
    "email": "contact@devfive.kr"
  },
  "keywords": [
    "css",
    "css-in-js",
    "css-in-js-preprocessor",
    "css-in-js-framework",
    "react"
  ],
  "version": "1.0.24",
  "type": "module",
  "scripts": {
    "lint": "eslint && tsc --noEmit",
    "build": "tsc && vite build"
  },
  "publishConfig": {
    "access": "public"
  },
  "sideEffects": false,
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    }
  },
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "dependencies": {
    "react": "^19.2",
    "csstype": "latest"
  },
  "devDependencies": {
    "rollup-plugin-preserve-directives": "^0.4",
    "vite": "^7.1",
    "vite-plugin-dts": "^4.5",
    "typescript": "^5.9",
    "@types/react": "^19.2",
    "@types/node": "^24.9"
  },
  "peerDependencies": {
    "react": "*",
    "csstype": "*"
  }
}
