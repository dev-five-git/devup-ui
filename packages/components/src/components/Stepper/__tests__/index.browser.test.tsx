import { fireEvent, render } from '@testing-library/react'

import {
  Stepper,
  StepperContainer,
  StepperDecreaseButton,
  StepperIncreaseButton,
  StepperInput,
} from '..'

describe('Stepper', () => {
  it('should render', () => {
    const { container } = render(
      <Stepper>
        <StepperContainer>
          <StepperDecreaseButton />
          <StepperInput />
          <StepperIncreaseButton />
        </StepperContainer>
      </Stepper>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should throw error if children are used outside of StepperProvider', () => {
    expect(() => {
      render(<StepperInput />)
    }).toThrow('useStepper must be used within a StepperProvider')
  })

  it('should call onValueChange when value is changed', () => {
    const onValueChange = vi.fn()
    const { container } = render(
      <Stepper onValueChange={onValueChange}>
        <StepperInput />
      </Stepper>,
    )
    const input = container.querySelector('[aria-label="Stepper value"]')
    fireEvent.change(input!, { target: { value: '10' } })
    expect(onValueChange).toHaveBeenCalledWith(10)
  })

  it('should change inner value when onValueChange is not provided', () => {
    const { container } = render(
      <Stepper>
        <StepperInput />
      </Stepper>,
    )
    const input = container.querySelector('[aria-label="Stepper value"]')
    fireEvent.change(input!, { target: { value: '10' } })
    expect(input).toHaveAttribute('data-value', '10')
  })

  it('should not change inner value by clicking the number when type is text', () => {
    const { container } = render(
      <Stepper type="text">
        <StepperInput />
      </Stepper>,
    )
    const input = container.querySelector('[aria-label="Stepper value"]')
    expect(input?.nodeName).toBe('DIV')
    expect(input).toHaveAttribute('data-value', '0')
  })

  it('should have disabled decrease button when value is at min', () => {
    const { container } = render(
      <Stepper>
        <StepperDecreaseButton />
        <StepperInput />
        <StepperIncreaseButton />
      </Stepper>,
    )
    const decreaseButton = container.querySelector(
      '[aria-label="Decrease button"] svg',
    )
    fireEvent.change(container.querySelector('[aria-label="Stepper value"]')!, {
      target: { value: '0' },
    })
    expect(decreaseButton).toHaveClass(
      'color-0-var(--base10,light-dark(#0000001A,#FFFFFF1A))--255',
    )
  })

  it('should have disabled increase button when value is at max', () => {
    const { container } = render(
      <Stepper>
        <StepperDecreaseButton />
        <StepperInput />
        <StepperIncreaseButton />
      </Stepper>,
    )
    const increaseButton = container.querySelector(
      '[aria-label="Increase button"] svg',
    )
    fireEvent.change(container.querySelector('[aria-label="Stepper value"]')!, {
      target: { value: '100' },
    })
    expect(increaseButton).toHaveClass(
      'color-0-var(--base10,light-dark(#0000001A,#FFFFFF1A))--255',
    )
  })

  it('should export components', async () => {
    const index = await import('../index')
    expect({ ...index }).toEqual({
      Stepper: expect.any(Function),
      StepperContainer: expect.any(Function),
      StepperDecreaseButton: expect.any(Function),
      StepperIncreaseButton: expect.any(Function),
      StepperInput: expect.any(Function),
      useStepper: expect.any(Function),
    })
  })

  it('should increase value when increase button is clicked', () => {
    const { container } = render(
      <Stepper>
        <StepperDecreaseButton />
        <StepperInput />
        <StepperIncreaseButton />
      </Stepper>,
    )
    const increaseButton = container.querySelector(
      '[aria-label="Increase button"]',
    )
    fireEvent.click(increaseButton!)
    expect(
      container.querySelector('[aria-label="Stepper value"]'),
    ).toHaveAttribute('data-value', '1')
  })

  it('should decrease value when decrease button is clicked', () => {
    const { container } = render(
      <Stepper defaultValue={1}>
        <StepperDecreaseButton />
        <StepperInput />
        <StepperIncreaseButton />
      </Stepper>,
    )
    const decreaseButton = container.querySelector(
      '[aria-label="Decrease button"]',
    )
    fireEvent.click(decreaseButton!)
    expect(
      container.querySelector('[aria-label="Stepper value"]'),
    ).toHaveAttribute('data-value', '0')
  })
})
