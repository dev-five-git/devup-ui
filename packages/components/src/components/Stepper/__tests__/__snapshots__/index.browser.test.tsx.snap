// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stepper > should render 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--1 gap-0-8px--1 "
  >
    <button
      aria-disabled="true"
      aria-label="Decrease button"
      class="padding-0-0--255 height-0-28px--255 width-0-28px--255 border-radius-0-4px--255 outline-0-2px solid-17005923944751620165-1 box-sizing-0-border-box--1 cursor-0-pointer--1 font-weight-0-700--1 outline-offset-0-2px--1 position-0-relative--1 transition-0-.25s--1 color-0-var(--text,#272727)-15425828959012638752-1 background-0-color-mix(in srgb,var(--primary,#8163E1) 20%,#FFF 80%)-15425828959012638752-1 border-0-1px solid var(--primary,#8163E1)-15425828959012638752-1 color-0-#D6D7DE-14172363753176421546-1 background-color-0-#F0F0F3-14172363753176421546-1 cursor-0-not-allowed-14172363753176421546-1 border-color-0-var(--border,#E4E4E4)-14172363753176421546-1 outline-color-0-var(--primaryFocus,#9385D3)-17005923944751620165-1 border-color-0-var(--primary,#8163E1)-8380715471663921674-1 background-0-color-mix(in srgb,var(--primary,#8163E1) 10%,#FFF 90%)-8380715471663921674-1 color-0-var(--text,#F6F6F6)-2922352740838246662-1 background-0-var(--primary,#8163E1)-2922352740838246662-1 color-0-#373737-878116160589243838-1 background-color-0-#47474A-878116160589243838-1 border-color-0-transparent-878116160589243838-1 border-color-0-var(--primary,#8163E1)-6232724021015440856-1 background-0-color-mix(in srgb,var(--primary,#674DC7) 10%,var(--inputBackground,#2E2E2E) 90%)-6232724021015440856-1 outline-color-0-var(--primaryFocus,#927CE4)-13318702800233181468-1 background-0-var(--inputBackground,#2E2E2E)-6667598448774358329-1 background-0-var(--inputBackground,#FFF)--1 border-0-1px solid var(--border,#E4E4E4)--1 border-radius-0-10px--1 color-0-var(--text,#272727)--1 font-size-0-14px--1 font-size-4-15px--1 letter-spacing-0--.02em--1 letter-spacing-4--.03em--1 padding-right-0-16px--1 padding-left-0-16px--1 padding-bottom-0-10px--1 padding-top-0-10px--1 "
      disabled=""
      type="button"
    >
      <div
        class="max-width-0-100%--255 margin-right-0-auto--255 margin-left-0-auto--255 position-0-relative--255 width-0-fit-content--255"
      >
        <div
          class=" line-height-0-1.2--255 min-height-0-1.2em--255 "
        >
          <svg
            class="color-0-var(--base10,light-dark(#0000001A,#FFFFFF1A))--255"
            fill="none"
            height="28"
            viewBox="0 0 28 28"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9 14C9 13.4477 9.3731 13 9.83333 13H18.1667C18.6269 13 19 13.4477 19 14C19 14.5523 18.6269 15 18.1667 15H9.83333C9.3731 15 9 14.5523 9 14Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </button>
    <div
      class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
    >
      <input
        aria-label="Stepper value"
        class="width-0-60px--2 text-align-0-center--2 border-radius-0-6px--2 display-0-none-1599573265477977541-2 undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-12px--1 "
        data-value="0"
        type="number"
        value="0"
      />
      0
    </div>
    <button
      aria-disabled="false"
      aria-label="Increase button"
      class="padding-0-0--255 height-0-28px--255 width-0-28px--255 border-radius-0-4px--255 outline-0-2px solid-17005923944751620165-1 box-sizing-0-border-box--1 cursor-0-pointer--1 font-weight-0-700--1 outline-offset-0-2px--1 position-0-relative--1 transition-0-.25s--1 color-0-var(--text,#272727)-15425828959012638752-1 background-0-color-mix(in srgb,var(--primary,#8163E1) 20%,#FFF 80%)-15425828959012638752-1 border-0-1px solid var(--primary,#8163E1)-15425828959012638752-1 color-0-#D6D7DE-14172363753176421546-1 background-color-0-#F0F0F3-14172363753176421546-1 cursor-0-not-allowed-14172363753176421546-1 border-color-0-var(--border,#E4E4E4)-14172363753176421546-1 outline-color-0-var(--primaryFocus,#9385D3)-17005923944751620165-1 border-color-0-var(--primary,#8163E1)-8380715471663921674-1 background-0-color-mix(in srgb,var(--primary,#8163E1) 10%,#FFF 90%)-8380715471663921674-1 color-0-var(--text,#F6F6F6)-2922352740838246662-1 background-0-var(--primary,#8163E1)-2922352740838246662-1 color-0-#373737-878116160589243838-1 background-color-0-#47474A-878116160589243838-1 border-color-0-transparent-878116160589243838-1 border-color-0-var(--primary,#8163E1)-6232724021015440856-1 background-0-color-mix(in srgb,var(--primary,#674DC7) 10%,var(--inputBackground,#2E2E2E) 90%)-6232724021015440856-1 outline-color-0-var(--primaryFocus,#927CE4)-13318702800233181468-1 background-0-var(--inputBackground,#2E2E2E)-6667598448774358329-1 background-0-var(--inputBackground,#FFF)--1 border-0-1px solid var(--border,#E4E4E4)--1 border-radius-0-10px--1 color-0-var(--text,#272727)--1 font-size-0-14px--1 font-size-4-15px--1 letter-spacing-0--.02em--1 letter-spacing-4--.03em--1 padding-right-0-16px--1 padding-left-0-16px--1 padding-bottom-0-10px--1 padding-top-0-10px--1 "
      type="button"
    >
      <div
        class="max-width-0-100%--255 margin-right-0-auto--255 margin-left-0-auto--255 position-0-relative--255 width-0-fit-content--255"
      >
        <div
          class=" line-height-0-1.2--255 min-height-0-1.2em--255 "
        >
          <svg
            class="color-0-var(--text,light-dark(#272727,#F6F6F6))--255"
            fill="none"
            height="28"
            viewBox="0 0 28 28"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.8333 9.83333C14.8333 9.3731 14.4602 9 14 9C13.5397 9 13.1666 9.3731 13.1666 9.83333V13.1667H9.8333C9.37307 13.1667 8.99997 13.5398 8.99997 14C8.99997 14.4602 9.37307 14.8333 9.8333 14.8333H13.1666V18.1667C13.1666 18.6269 13.5397 19 14 19C14.4602 19 14.8333 18.6269 14.8333 18.1667V14.8333H18.1666C18.6269 14.8333 19 14.4602 19 14C19 13.5398 18.6269 13.1667 18.1666 13.1667H14.8333V9.83333Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;
